microsoft-edge/extensions-chromium/developer-guide/minimize-page-load-time-impact.md

```diff
@@ -0,0 +1,251 @@
+---
+title: Minimize your extension's impact on page load time
+description: Best Practices to minimize the impact your extension has on the page load time.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.service: microsoft-edge
+ms.subservice: extensions
+ms.date: 09/18/2024
+---
+# Minimize your extension's impact on page load time
+
+Content scripts are JavaScript files that your extension injects into webpages, and that run in the context of these webpages. Using content scripts, your extension can access and modify a rendered webpage by reading from or changing the DOM.
+
+Content scripts can, however, have a noticeable impact on the performance of a webpage, for example by slowing down the page load time. This can happen if the content script runs a lot of code while the page is loading.
+
+This article provides best practices that can help you minimize the performance impact of your extension on the webpages that your users visit.
+
+
+<!-- ====================================================================== -->
+## Profile your extension content script
+
+To profile the performance of your extension's content script, use Microsoft Edge DevTools or the Edge tracing tool, as described in the following sections.
+
+
+<!-- ------------------------------ -->
+#### Profile your content script using Microsoft Edge DevTools
+
+DevTools provides a set of features for inspecting, debugging, and profiling the code that a webpage uses. DevTools can also be used to profile your extension's code.
+
+In this section, you'll learn how to use the **Performance** tool in DevTools to profile your extension's content script. To learn more about the **Performance** tool, see [Introduction to the Performance tool](../../devtools-guide-chromium/evaluate-performance/index.md).
+
+1. To open DevTools, right-click the webpage, and then select **Inspect**.  Or, press **Ctrl+Shift+I** (Windows, Linux) or **Command+Option+I** (macOS).  DevTools opens.
+
+1. In DevTools, on the **Activity Bar**, select the **Performance** (![Performance tool icon](./minimize-page-load-time-impact-images/performance-tool-icon.png)) tab.  If that tab isn't visible, select **More tools** (![More tools icon](./minimize-page-load-time-impact-images/more-tools-icon.png)) > **Performance**.
+
+1. To start recording a performance profile, click **Record** (![Record icon](./minimize-page-load-time-impact-images/record-icon.png)).
```

Comment:
```suggestion
1. To start recording a performance profile, click the **Record** (![Record icon](./minimize-page-load-time-impact-images/record-icon.png)) button.
```
corp style: "don't mention the UI control type, unless it clarifies", & "don't assume the user can see pngs"
response: it always clarifies.  paint mental picture of UI via words

I plan to push commits for writer/editor pass